source/renderer/app/components/assets/AssetAmount.js

```diff
@@ -26,47 +30,54 @@ type Props = {
   isShort?: boolean,
 };
 
-@observer
-export default class AssetAmount extends Component<Props> {
-  static contextTypes = {
-    intl: intlShape.isRequired,
-  };
+function AssetAmount({
+  amount,
+  metadata,
+  decimals,
+  isLoading,
+  className,
+  isShort,
+}: Props) {
+  const discreetModeFeature = useDiscreetModeFeature();
 
-  render() {
-    const {
-      amount,
-      metadata,
-      decimals,
-      isLoading,
-      className,
-      isShort,
-    } = this.props;
-    if (isLoading) return '-';
-    const componentStyles = classnames([styles.component, className]);
-    const content = !isLoading
-      ? formattedTokenWalletAmount(amount, metadata, decimals, isShort)
-      : '-';
-    return (
-      <div className={componentStyles}>
-        {decimals ? (
-          <PopOver
-            content={
-              <FormattedHTMLMessage
-                {...messages.unformattedAmount}
-                values={{
-                  amount: formattedTokenWalletAmount(amount, null, 0),
-                }}
-              />
-            }
-            visible={decimals ? undefined : false}
-            className={styles.unformattedAmount}
-          >
+  if (isLoading) return '-';
+  const componentStyles = classnames([styles.component, className]);
+  const content = !isLoading
+    ? formattedTokenWalletAmount(amount, metadata, decimals, isShort)
+    : '-';
+
+  const { ticker } = metadata || {};
+
+  return (
+    <div className={componentStyles}>
+      {decimals ? (
+        <PopOver
+          content={
+            <FormattedHTMLMessage
+              {...messages.unformattedAmount}
+              values={{
+                amount: discreetModeFeature.hideSensitiveData(
+                  formattedTokenWalletAmount(amount, null, 0)
+                ),
+              }}
+            />
+          }
+          visible={decimals ? undefined : false}
+          className={styles.unformattedAmount}
+        >
+          <DiscreetValue ticker={{ show: Boolean(ticker), symbol: ticker }}>
```

Comment:
Hm, i think we should either get rid of the `ticker` property of this component OR rename it to `CurrencyValue` and rename `ticker` to `currency` (because it could also be EUR or USD) 

its name suggests that this could be any value, not just coin ticker values so this API design does not fit IMO. Btw is it really necessary to provide the ticker value to this component? why not just wrap the number?